tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/AsyncControllerActionInvoker_EndInvokeAction_Integration.cs

```diff
@@ -95,7 +95,18 @@ public class AsyncControllerActionInvoker_EndInvokeAction_Integration
         private static void OnRequestCompleted(HttpContext httpContext, Scope scope, DateTimeOffset finishTime)
         {
             HttpContextHelper.AddHeaderTagsFromHttpResponse(httpContext, scope);
-            scope.Span.SetHttpStatusCode(httpContext.Response.StatusCode, isServer: true, Tracer.Instance.Settings);
+
+            if (!HttpRuntime.UsingIntegratedPipeline && httpContext.Response.StatusCode == 200)
+            {
+                // in classic mode, the exception won't cause the correct status code to be set
+                // even though a 500 response will be sent, so set it manually instead
+                scope.Span.SetHttpStatusCode(500, isServer: true, Tracer.Instance.Settings);
```

Comment:
How do we know an exception happened in this case? Looks like we're only checking for classic mode and status code 200.

Reply:
Never mind, found it above when I looked at the complete file :)

Reply:
> Consider renaming `OnRequestCompleted()` to make it clear that it's only called when an exception happens

Yeah, I thought the same thing tbh, will do 👍

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2854/commits/16df505d69d591ea8d33c872cac56eb2a9f967ab (`OnRequestCompleted -> OnRequestCompletedAfterException`)